### ISSUE TITLE
docs: Testing-Your-Application

### ISSUE THREAD

**Issue by bajtos** [strongloop/loopback-next#553]
## Overview
Review the content of http://loopback.io/doc/en/lb4/Testing-your-application.html and update it to match the current version of http://loopback.io/doc/en/lb4/Thinking-in-LoopBack.html

## Acceptance Criteria
**In LoopBack.io**:
- Rewrite "Testing Your Application" to reflect the current design of loopback-next alpha, and to use the approach outlined in "Thinking in LoopBack".
- Split the content of "Testing Your Application" into two parts:
  - "High level" items should move to "Thinking in LoopBack"
  - "Details" should remain in the "Testing Your Application" page
- Dually link both pages to one another (to both give people a high-level overview of _how_ testing should be implemented in their application, and exactly _what_ is required to make it happen)
- Review the following list, and create a story to add what's missing to "Testing your Application":
  - test your api
  - test your sequence
  - test your controllers
  - test your repositories
  - test your services
  - validate API specs
  - create mocks
  - create stubs
  - create spys

## Notes

There is a PR in loopback-next-hello-world (https://github.com/strongloop/loopback-next-hello-world/pull/20) that you can update to validate the existing snippets used in Thinking-in-LoopBack.html (as well as test any changes you need to make to the docs!)

**Comment by crandmck** [comment]
Also, review https://github.com/strongloop/loopback-next/wiki/Testing-Your-Application-%28Advanced%29 and move any useful info into https://github.com/strongloop/loopback-next/wiki/Testing-Your-Application.

**Comment by bajtos** [comment]
This is blocked by https://github.com/strongloop/loopback.io/pull/475, which is making quite a lot of changes in pages we are going to rework.

**Comment by bajtos** [comment]
> test your services

Let's write this content as part of "Guide: Services" - see #451.

**Comment by bajtos** [comment]
Another follow-up story: https://github.com/strongloop/loopback-next/issues/648 Write a better advice on testing custom sequences
